index.js

```diff
@@ -1,106 +1,118 @@
-'use strict'
-const dns = require('dns')
-const setConfig = require('./src/config.js')
-const Context = require('./src/context.js')
-const Callback = require('./src/callback.js')
-const Report = require('./src/report.js')
-const globals = require('./src/globals')
+'use strict';
+const dns = require('dns');
+const setConfig = require('./src/config.js');
+const Context = require('./src/context.js');
+const Callback = require('./src/callback.js');
+const Report = require('./src/report.js');
+const globals = require('./src/globals');
 
 function makeDnsPromise(host) {
   return new Promise((resolve, reject) => {
     dns.lookup(host, (err, address) => {
       if (err) {
-        reject(err)
+        reject(err);
       }
-      resolve(address)
-    })
-  })
+      resolve(address);
+    });
+  });
 }
 
 function setupTimeoutCapture(config, report, context) {
+  var endTime = 599900; /* Maximum execution: 100ms short of 5 minutes */
   if (config.timeoutWindow < 1) {
-    return
+    return undefined;
   }
 
-  var endTime = 599900  /* Maximum execution: 100ms short of 5 minutes */
   if (config.timeoutWindow > 0 && context && context.getRemainingTimeInMillis) {
-    endTime = Math.max(0, context.getRemainingTimeInMillis() - config.timeoutWindow)
+    endTime = Math.max(
+      0,
+      context.getRemainingTimeInMillis() - config.timeoutWindow
+    );
   }
 
   return setTimeout(() => {
-    report.send(new Error("Timeout Exceeded."), function noop() {})
-  }, endTime)
+    report.send(new Error('Timeout Exceeded.'), function noop() {});
+  }, endTime);
 }
 
-module.exports = function(options) {
-  var fn = function(func) {
-    fn.metricsQueue = []
-    const config = setConfig(options)
+module.exports = options => {
+  const fn = func => {
+    var dnsPromise = undefined;
+    fn.metricsQueue = [];
+    const config = setConfig(options);
 
     if (!config.clientId) {
       // No-op if user doesn't set an IOpipe token.
-      return func
+      return func;
     }
 
     /* resolve DNS early on coldstarts */
-    var dnsPromise = makeDnsPromise(config.host)
+    dnsPromise = makeDnsPromise(config.host);
 
-    return function() {
-      fn.metricsQueue = []
-      var args = [].slice.call(arguments)
+    return function IOpipeWrapper() {
+      fn.metricsQueue = [];
+      const args = [].slice.call(arguments);
```

Comment:
An array being object-like, `const` is ok. You can change individual indices without errors being thrown. My thought is we should use var as only last resort when the parent object will get replaced. If you want to stay strong on the var here, I can deal